packages/maria/src/lib/MariaProvider.ts

```diff
@@ -517,60 +517,70 @@ export class MariaProvider<StoredValue = unknown> extends JoshProvider<StoredVal
   }
 
   public async [Method.Random](payload: Payload.Random<StoredValue>): Promise<Payload.Random<StoredValue>> {
-    const { count, duplicates } = payload;
+    const { count, unique } = payload;
     const size = await this.handler.size();
 
-    if (size === 0) return { ...payload, data: [] };
-    if (size < count) {
+    if (unique && size < count) {
       payload.errors.push(this.error({ identifier: CommonIdentifiers.InvalidCount, method: Method.Random }));
 
       return payload;
     }
 
+    if (size === 0) {
+      payload.errors.push(this.error({ identifier: CommonIdentifiers.MissingData, method: Method.Random, context: { unique, count } }));
+
+      return payload;
+    }
+
     payload.data = [];
 
     const keys = await this.handler.keys();
 
-    if (duplicates) {
-      while (payload.data.length < count) {
-        const key = keys[Math.floor(Math.random() * size)];
-
-        payload.data.push((await this.handler.get(key))!);
-      }
-    } else {
+    if (unique) {
       const randomKeys = new Set<string>();
```

Comment:
Does mariadb have any way to query random that would be faster than this?

Reply:
Looks like it does have some kind of random query but its somehow worse than ours 
https://mariadb.com/kb/en/rand/
> Note, however, that this technique should never be used on a large table as it will be extremely slow. MariaDB will read all rows in the table, generate a random value for each of them, order them, and finally will apply the LIMIT clause.

Reply:
Can we test this, it's possible while not the best it's better than the current solution

Reply:
I mean you can. My Maria doesn't work. 
However just based off that description, it sounds to me like that is slower. It has to fetch all the keys, generate a random value for each key in the table, sort them and then choose the top x
Whereas ours is fetch all the keys, and repeat the amount of times we need to get, picking a random index